### DISCUSSION TITLE
Implementing semantic release to an on-going project

### DISCUSSION THREAD

**Discussion by muratcorlu** [semantic-release/semantic-release#2437] [Configuration] [answered]
[We have a JS project](https://github.com/trendyol/grace) that has already been published to NPM in v1.0.39 **without semantic-release**.

Now with v2, we want to start using semantic-release. Our v2 work is in [`next` branch](https://github.com/Trendyol/grace/tree/next). We want to start publishing works in this branch as beta versions of v2. Here is our current `.releaserc` file:

```json
{
  "branches": [
    "main",
    { "name": "1.0.x", "range": "1.0.x", "channel": "1.0.x" },
    {
      "name": "next",
      "prerelease": "beta",
      "channel": "beta"
    }
  ],
  "plugins": [
    "@semantic-release/commit-analyzer",
    "@semantic-release/release-notes-generator"
  ]
}
```

When I run `npx semantic-release --dry-run` in my local within `next` branch I see:

```
› ℹ  There is no previous release, the next release version is 1.0.0-beta.1
```

[I tagged 2 commits](https://github.com/Trendyol/grace/tags) in `next` branch as `v2.0.0-beta.0` and `v2.0.0-beta.1` manually to set an initial version for semantic-release but that didn't help.

What should I do to start making beta releases with pushes to `next` branch?

**Comment by travi** [comment, marked as answer]
you likely need to set the git notes, as described in https://semantic-release.gitbook.io/semantic-release/support/troubleshooting#release-not-found-release-branch-after-git-push-force
